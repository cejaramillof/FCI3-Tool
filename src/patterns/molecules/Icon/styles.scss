@import "./../../../globalStyles/variables";

.icon-competence {

    &.icon-badge {

        &-s {
            border-radius: 14px;
            padding: 4px;

            img {
                height: 14px;
            }
        }

        &-m {
            border-radius: 24px;
            padding: 6px;

            img {
                height: 24px;
            }
        }

        &-l {
            border-radius: 34px;
            padding: 8px;

            img {
                height: 34px;
            }
        }
    }

    &__need-work-on-this{
        background: $needs-work-on-this;
    }

    &__cool{
        background: $cool;
    }

    &__you-rock{
        background: $you-rock;
    }
}