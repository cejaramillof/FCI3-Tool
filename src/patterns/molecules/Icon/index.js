import React from 'react';
import IconCoolResource from 'img/icons/cool.jpeg';
import IconNeedWorkOnThisResource from 'img/icons/need-work-on-this.jpeg';
import IconYouRockResource from 'img/icons/you-rock.jpeg';
import Image from 'patterns/atoms/Image';
import util from 'lib/utils';
import './styles.scss';

const Icon = ({cssIcon, size = 's'}) => {
    const icons = {
        'cool': IconCoolResource,
        'need-work-on-this': IconNeedWorkOnThisResource,
        'you-rock': IconYouRockResource
    };

    const getAltText = () => {
        return cssIcon &&  icons[cssIcon] ?
            util.capitalize(cssIcon).replace('-', ' ') : '';
    };

    return (
        <span
            className={`badge badge-light icon-competence icon-competence__${cssIcon} icon-badge-${size}`}
            >
            <Image image={icons[cssIcon]} alt={getAltText()} />
        </span>
    )
};

export default Icon;