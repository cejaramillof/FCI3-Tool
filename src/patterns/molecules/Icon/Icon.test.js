import React from 'react';
import { mount } from 'enzyme';
import Icon from './index';

const props = {
  cssIcon: 'cool',
  size: 's'
};

const badProps = {
    cssIcon:'super test'
};

it('should render correctly', () => {
    const wrapper = mount( <Icon {...props} /> );

    expect(wrapper.find('span').children().name()).toBe('Image');
    expect(wrapper.find('span').children().props()).toEqual({
        alt: 'Cool',
        image: 'cool.jpeg'
    });
});

describe('when bad props are provided', () => {
    const wrapper = mount( <Icon {...badProps} /> );

    it('should render properly', function () {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.props()).toEqual(badProps);
        expect(wrapper.find('span').children().name()).toBe('Image');
        expect( wrapper.find('span').children().props()).toEqual({
            alt: '',
            image: undefined
        });
    });

    it('should get blank alt text when cssIcon is not provided', ()  => {
        wrapper.setProps({cssIcon: null})
        expect(wrapper.find('span').children().props()).toEqual({
            alt: '',
            image: undefined
        });
    })
});

