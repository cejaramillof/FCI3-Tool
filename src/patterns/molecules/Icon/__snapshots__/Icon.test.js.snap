// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when bad props are provided should render properly 1`] = `ReactWrapper {}`;
